project/src/main/webapp/script.js

```diff
@@ -62,11 +121,131 @@ function createPlaceElement(place) {
 }
 
 /**
- * Displays the query form to the user and hides the results, so that the user can try again with a
- * different query.
+ * Displays the query form to the user and hides the results, so that the user can try to get new
+ * results.
  */
 function tryAgain() {
   document.getElementById('query-form').style.display = 'block';
   document.getElementById('results').style.display = 'none';
   document.getElementById('place').innerHTML = '';
 }
+
+function addSearchBoxToMap(map, input) {
```

Comment:
I don't quite understand what this function is doing, can you add a comment?
Is this the function that gets the location from the user and displays it on the map letting the user change it through the searchbox? If it is, why is it handeling a list of markers and places and not just one place the user enters? I mean in what situation this map handels more than one place and one marker shown on it?

Reply:
This is not the function that gets the user's device location, it adds a search box to the map and defines everything that the search box needs to be able to do. I added a comment. 
The situation is for example when the user types something and presses "enter" without choosing a specific location, then the locations that fit that query the most are shown to him (as markers) and he can choose from them.

Reply:
Oh ok, I didn't understand that has the functionality of showing several optional locations, now it makes more sense.